src/main/java/com/fasterxml/jackson/databind/jsontype/TypeResolverBuilder.java

```diff
@@ -97,6 +98,21 @@ public TypeDeserializer buildTypeDeserializer(DeserializationConfig config,
      *   but not necessarily)
      */
     public T init(JsonTypeInfo.Id idType, TypeIdResolver res);
+    
+    /**
+     * Initialization method that is called right after constructing
+     * the builder instance, in cases where information could not be
+     * passed directly (for example when instantiated for an annotation)
+     *
+     * @param settings Configuration settings to apply.
+     *
+     * @return Resulting builder instance (usually this builder,
+     *   but not necessarily)
+     *   
+     * @since 2.16
+     */
+    public T init(JsonTypeInfo.Value settings, TypeIdResolver res);
```

Comment:
Would require default implementation, I think, to avoid breakage wrt backwards-compatibility (existing implementations can't really implement it). Not sure if that impl should do nothing or throw an exception...